src/components/Wizard/InfoBarWizard/index.tsx

```diff
@@ -43,18 +44,50 @@ const PRIORITIES = [
 interface InfoBarWizardProps {
   id: string;
   stopEditing: () => void;
+  setPreviewJson: (json: object | undefined) => void;
 }
 
-export default function InfoBarWizard({ id, stopEditing }: InfoBarWizardProps) {
+export default function InfoBarWizard({
+  id,
+  stopEditing,
+  setPreviewJson,
+}: InfoBarWizardProps) {
   const formContext = useForm<InfoBarWizardFormData>();
-  const { register, watch, trigger } = formContext;
+  const { register, watch, trigger, getValues } = formContext;
   const priority = watch("content.priority") ?? { enabled: false, value: 0 };
 
+  const validate = async (): Promise<object | null> => {
+    if (await trigger(undefined, { shouldFocus: true })) {
+      return formDataToMessageJson(id, getValues());
+    }
+
+    return null;
+  };
+
   const handleShowJson = async (): Promise<void> => {
-    await trigger(undefined, { shouldFocus: true });
+    try {
+      const json = await validate();
+      if (json) {
+        setPreviewJson(json);
+      }
+    } catch (e) {
+      console.error(e);
+    }
   };
 
-  const handlePreview = handleShowJson;
+  const handlePreview = async (): Promise<void> => {
+    try {
+      const json = await validate();
+      if (json) {
+        const jsonStr = JSON.stringify(json);
+        const uri = `about:messagepreview?json=${btoa(jsonStr)}`;
+
+        return navigator.clipboard.writeText(uri);
```

Comment:
This works great - might be nice to give the button `cursor: copy;` or something similar to indicate it's going to copy the URL to clipboard (instead of just opening the preview which I think folks might assume)

Reply:
TIL about `cursor: copy`.

I put this on both copy buttons.